todo_app/data/trello.py

```diff
@@ -0,0 +1,118 @@
+from enum import Enum
+from typing import List
+
+import requests
+
+from todo_app.data.item import Item
+
+
+class CardStatus(Enum):
+    DONE = 'Done'
+    TODO = 'To Do'
+
+    @classmethod
+    def get_status(cls, list_id: str, board_lists: List[dict]) -> Enum:
+        status = next((list['name'] for list in board_lists if list['id'] == list_id), None)
+        if status == cls.DONE.value:
+            return cls.DONE
+        elif status == cls.TODO.value:
+            return cls.TODO
+        else:
+            raise ValueError('Unknown status')
+
+
+class Trello:
+    BASE_URL = "https://api.trello.com/1/"
+
+    def __init__(self, app_key: str, token: str, board_id: str) -> None:
+        self.app_key = app_key
+        self.token = token
+        self.board_id = board_id
+        self.board_lists = self.get_board_lists()
+
+    def get_items(self) -> List[dict]:
+        """
+        Fetches all saved items from Trello. 
+
+        Returns:
+            list: The list of saved items.
+        """
+        endpoint = f"boards/{self.board_id}/cards"
+        params = {
+            'key': self.app_key,
+            'token': self.token,
+        }
+
+        cards = []
+        json_response = requests.get(self.BASE_URL + endpoint, params=params).json()
+        for card in json_response:
+            id = card['id']
+            title = card['name']
+            description = card['desc']
+            list_id = card['idList'] 
+            status = CardStatus.get_status(list_id, self.board_lists)
+            cards.append(Item(id, title, description, status))
+
+        sorted_cards = sorted(cards, key=lambda card: card.status.value, reverse=True) 
+        return sorted_cards
+
+    def get_board_lists(self) -> List[dict]:
+        endpoint = f"boards/{self.board_id}/lists"
+        params = {
+            'key': self.app_key,
+            'token': self.token,
+        }
+        resp = requests.get(self.BASE_URL + endpoint, params=params)
+        return resp.json()
```

Comment:
Since we store this result directly on our Trello object, it would be nice to convert this into a known class before we store it (e.g. convert the response into an array of a new `List` class